Tests/NIOHTTP2Tests/TestUtilities.swift

```diff
@@ -64,6 +64,8 @@ extension XCTestCase {
 
     /// Have two `NIOAsyncTestingChannel` objects send and receive data from each other until
     /// they make no forward progress.
+    ///
+    /// ** This function is not thread safe and can lead to deadlocks, prefer the one-way variant which is less error-prone**
```

Comment:
It _is_ thread-safe, the problem is that calling it from multiple threads is racy.